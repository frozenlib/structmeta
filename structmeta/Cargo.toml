[package]
name = "structmeta"
version = "0.1.0"
authors = ["frozenlib"]
license = "MIT OR Apache-2.0"
readme = "../README.md"
repository = "https://github.com/frozenlib/structmeta"
documentation = "https://docs.rs/structmeta/"
keywords = ["derive", "parse", "attribute", "syn", "totokens"]
categories = ["development-tools::procedural-macro-helpers"]
description = "Parse Rust's attribute arguments by defining a struct."
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
structmeta-derive = { version = "=0.1.0", path = "../structmeta-derive" }
proc-macro2 = "1.0.24"
syn = "1.0.65"
quote = "1.0.9"
