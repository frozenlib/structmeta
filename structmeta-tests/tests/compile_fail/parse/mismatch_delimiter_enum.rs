use structmeta::Parse;
use syn::LitStr;
#[derive(Parse)]
enum TestType {
    A {
        #[to_tokens("(")]
        braket_token: syn::token::Bracket,
        str: LitStr,
    },
    B(LitStr),
}

fn main() {}
