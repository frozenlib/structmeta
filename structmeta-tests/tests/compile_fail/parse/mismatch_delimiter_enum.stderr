error[E0308]: mismatched types
 --> tests/compile_fail/parse/mismatch_delimiter_enum.rs:7:9
  |
7 |         braket_token: syn::token::Bracket,
  |         ^^^^^^^^^^^^  ------------------- expected due to this
  |         |
  |         expected struct `Bracket`, found struct `Paren`
